drivers/net/ethernet/renesas/rswitch.c

```diff
@@ -3848,9 +3983,7 @@ static int renesas_eth_sw_probe(struct platform_device *pdev)
 
 	/* Fixed to use GWCA1 */
 	priv->gwca.index = 4;
-	priv->gwca.num_chains = num_ndev * NUM_CHAINS_PER_NDEV +
-		num_virt_devices * 2 * NUM_CHAINS_PER_NDEV;
-
+	priv->gwca.num_chains = RSWITCH_MAX_NUM_CHAINS;
```

Comment:
As I know, we have an issue with using chain numbers of more than 31. In case of adding a lot of vlan devices, this issue (of course if it remains) can occur. We have to recheck this.